### ISSUE TITLE
Label rendering glitch using Pygame / Unicode text / Custom font

### ISSUE THREAD

**Issue by tito** [kivy/kivy#2723]
Simple text case, if you render a label with "Hello " (note the space at the end), some font will display a box. The new changes in master replace the space before/after the text with unbreakable space unicode. But they don't replace back with standard space when rendering, which give us a beautiful box :)

This can be seen with a font like MrEavesSanOT-Reg.otf (http://www.myfonts.com/fonts/emigre/mr-eaves-sans-opentype/ot-reg/specs.html)

The current fixes i did in pygame is:

```
diff --git a/kivy/core/text/text_pygame.py b/kivy/core/text/text_pygame.py
index abbf2dc..7b7a58a 100644
--- a/kivy/core/text/text_pygame.py
+++ b/kivy/core/text/text_pygame.py
@@ -92,6 +92,7 @@ class LabelPygame(LabelBase):
         self._pygame_surface.fill((0, 0, 0, 0))
 
     def _render_text(self, text, x, y):
+        text = text.replace(u"\u00a0", " ")
         font = self._get_font()
         color = [c * 255 for c in self.options['color']]
         color[0], color[2] = color[2], color[0]
```

But i guess that was not the right place to put it. Any advice? @matham ?

**Comment by kived** [comment]
So, I think we should either:

1) Merge my PR, close this issue, and open another issue to review this for 1.9.1.
2) Revert my previous PR which caused this issue, reopen the previous issue that PR fixed, and tag it 1.9.1.
3) Someone (@matham?) submit a replacement for my previous PR and merge it today.

**Comment by matham** [comment]
I'll look into adding a variable called `internal_strip`, which will allow not stripping only internal spaces.

**Comment by matham** [comment]
It's actually quite difficult to do this with markup, because it's not straight forward to figure out what is the before/after whitespace, since there could be markup text.

But also, I'm not sure that always stripping internal text is always correct. E.g. when `size_hint_x` is `None`, the user knows exactly how it'd layout each line. So I'm wondering whether we should not go back to how it was and instead fix the original problem.

**Comment by matham** [comment]
This should now be fixed with #2840.

**Comment by akshayaurora** [comment]
closed  by matham in #2840
